llvm/test/CodeGen/NVPTX/rotate_64.ll

```diff
@@ -6,19 +6,19 @@ declare i64 @llvm.nvvm.rotate.right.b64(i64, i32)
 
 ; CHECK: rotate64
 define i64 @rotate64(i64 %a, i32 %b) {
-; CHECK: shl.b64         [[LHS:%.*]], [[RD1:%.*]], 3;
-; CHECK: shr.b64         [[RHS:%.*]], [[RD1]], 61;
-; CHECK: add.u64         [[RD2:%.*]], [[LHS]], [[RHS]];
+; CHECK: shr.u64         [[RHS:%.*]], [[RD1:%.*]], 61;
```

Comment:
This test should probably be converted to use automatic check generation, too.
One of the cases where we may have spotted wrong operation order with all instructions visible.

Reply:
Updated